### PR TITLE
Replace PublicUserType with models.PublicUser type

### PR DESCRIPTION
**Pull request by ajlacey**
This PR uses the new PublicUser type from the model in the user handlers.

### CODE REVIEW

File: backend/handlers/markets/marketdetailshandler.go [hunk 1]
```diff
@@ -55,10 +55,6 @@ func MarketDetailsHandler(w http.ResponseWriter, r *http.Request) {
 
 	// find the number of users on the market
 	numUsers := models.GetNumMarketUsers(bets)
-	if err != nil {
-		http.Error(w, "Error retrieving number of users.", http.StatusInternalServerError)
-		return
-	}
```
Comment (ajlacey) [inline backend/handlers/markets/marketdetailshandler.go:61]: There are no errors currently returned by the call to `GetNumMarketUsers(bets)`

File: backend/handlers/users/publicuser.go [hunk 2]
```diff
@@ -10,21 +10,6 @@ import (
...
-	InitialAccountBalance int64  `json:"initialAccountBalance"`
-	AccountBalance        int64  `json:"accountBalance"`
-	PersonalEmoji         string `json:"personalEmoji,omitempty"`
-	Description           string `json:"description,omitempty"`
-	PersonalLink1         string `json:"personalink1,omitempty"`
-	PersonalLink2         string `json:"personalink2,omitempty"`
-	PersonalLink3         string `json:"personalink3,omitempty"`
-	PersonalLink4         string `json:"personalink4,omitempty"`
-}
-
```
Comment (ajlacey) [inline backend/handlers/users/publicuser.go:27]: All of these fields are just being re-declared from the model. By making them a component type of the model User, we can access them all together.

**Review by j4qfrost — APPROVED** [review]
LTGM, just a lift and shift right?

**Comment by ajlacey** [comment]
> LTGM, just a lift and shift right?

Yeah, this is just how the changes to the model in #354 are going to be used.

**Review by pwdel — APPROVED** [review]
Tested on local, read code, makes sense, good consolidation.

**Pull request merged**